imagescan/delta.go

```diff
@@ -367,6 +332,12 @@ func (d *deltaState) isHostFsDisabled() bool {
 	return d.hostFSDisabled
 }
 
+func (d *deltaState) setImageScanned(key string) {
+	if img, found := d.images[key]; found {
+		img.scanned = true
```

Comment:
Not a fan of this boolean switch. Maybe it would better to have two maps (one for scanned images and one for pending images) and transfer images from one map to another when state changes? In that case addition to both maps would be idempotent and it would be easy to debug these maps